src/main/webapp/scripts/analyze.js

```diff
@@ -0,0 +1,44 @@
+function analyzeVideo() {
+  const videoId = document.getElementById('video-url').value.split('v=')[1];
+  const baseUrl = window.location.origin;
+  let url = new URL('/sentiment', baseUrl);
+  url.searchParams.append('video-id', videoId);
+
+  fetch(url)
+  .then(response => {
+    if (response.status >= 200 && response.status <= 299) {
+      return response.json();
+    } else if (response.status >= 400 && response.status <= 499) {
+      throw InvalidVideoException(response.statusText);
+    } else {
+      throw ServletFailureException(response.statusText);
+    }
+  })
+  .then(videoAnalysis => {
+    if (videoAnalysis.dataAvailable) {
+      updateSentimentContainer(videoAnalysis.score);
+    } else {
+      updateSentimentContainer("We couldn't analyze this video! Try again.")
```

Comment:
I believe the dataAvailable field is to tell the caller that there is no score. Maybe a different variable name would imply that better.